### PR TITLE
Delete bad users, add constraint MODPERMS-177

### PR DESCRIPTION
**Pull request by adamdickmeiss**
Try to keep the table name definitions in one place at least in Java.

The code that saves bad users is not yet discovered. This is just a repair, that will make it harder, but not impossible
to create bad users again (for example a JSONB with userId and additional property such as tags).

### CODE REVIEW

**Comment by adamdickmeiss** [comment]
The SQ checks is on old code as far I can tell. Should be dealt with separately, I think.

File: src/test/java/org/folio/permstest/RestVerticleTest.java [hunk 1]
```diff
@@ -3119,4 +3121,11 @@ public void permsUsersOffsetLimit(TestContext context) {
     context.assertEquals(200, response.code);
     context.assertEquals(3, response.body.getJsonArray("permissionUsers").size());
   }
+
+  @Test
+  public void testUserIdNull(TestContext context) {
+    PermissionUser u = new PermissionUser().withId(UUID.randomUUID().toString());
+    PostgresClient.getInstance(vertx, "diku").save(PermsAPI.TABLE_NAME_PERMSUSERS, u)
+        .onComplete(context.asyncAssertFailure(e -> context.assertTrue(e.getMessage().contains("permissions_users_userid_not_null"))));
```
Comment (julianladisch) [inline src/test/java/org/folio/permstest/RestVerticleTest.java:3129]:
```suggestion
        .onComplete(context.asyncAssertFailure(e -> assertThat(e.getMessage(), containsString("permissions_users_userid_not_null"))));
```

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Review by hjiebsco — APPROVED** [review]
LGTM

**Pull request merged**